src/Connect.svelte

```diff
@@ -10,19 +10,26 @@
 
   let walletUnavailable = !window.ethereum;
 
+  const onClickConnect = () => {
+    state.connectWalletConnect(config);
+  };
+  $: modalConnected = $store.status === ModalStateType.Open;
   $: connecting = $state.connection === Connection.Connecting;
 </script>
 
 <style>
 </style>
 
 <button
-  on:click={() => state.connect(config)}
+  on:click={onClickConnect}
   {style}
   class="connect {className}"
   disabled={connecting || walletUnavailable}
   data-waiting={connecting || null}
 >
+  {#if modalConnected}
+    <ModalWalletQRCode {config} uri={$store.modalProps?.uri || ''} />
```

Comment:
In what cases can `modalProps` be null? What does it mean for `ModalWalletQRCode` to be initialized without a uri?

Reply:
it can never be initialized with null. I set that because when the modalProps type won't exist on ConnectionModalState when the ConnectionModalStateType is set to close

Reply:
Hmm I see. Can we add that to the `if` condition instead then, somehow? The QRCode will not work properly with an empty URI, but this code makes it look like it's ok to pass "".

Reply:
Unresolved :warning: